index.js

```diff
@@ -180,72 +180,91 @@
   //  Value :: Location
   var Value = 'Value';
 
-  //  _funcPath :: (Boolean, Array String, a) -> Nullable Function
-  function _funcPath(allowInheritedProps, path, _x) {
-    var x = _x;
-    for (var idx = 0; idx < path.length; idx += 1) {
-      var k = path[idx];
-      if (x == null || !(allowInheritedProps || has(k, x))) return null;
-      x = x[k];
-    }
-    return typeof x === 'function' ? x : null;
+  //  prefix :: String
+  var prefix = 'fantasy-land/';
+
+  //  funcProp :: (String, a) -> Nullable Function
+  function funcProp(prop, x) {
+    return x != null && typeof x[prop] === 'function' ? x[prop] : null;
   }
 
-  //  funcPath :: (Array String, a) -> Nullable Function
-  function funcPath(path, x) {
-    return _funcPath(true, path, x);
+  //  functionImpl :: (String, String) -> Nullable Function
+  function functionImpl(type, name) {
+    if (implementations[type] == null) return null;
+    return implementations[type][name] || null;
   }
 
-  //  implPath :: Array String -> Nullable Function
-  function implPath(path) {
-    return _funcPath(false, path, implementations);
+  //  methodImpl :: (String, String) -> Nullable Function
+  function methodImpl(type, name) {
+    if (implementations[type] == null) return null;
```

Comment:
It throws an exception. This would happen if somebody were to construct a pretty silly argument:

```js
> Z.map(x => x, {constructor: {'@@type': 'toString'}})
TypeError: Cannot read property 'fantasy-land/map' of undefined
```